lib/nimble_publisher.ex

```diff
@@ -38,22 +38,35 @@ defmodule NimblePublisher do
     paths = from |> Path.wildcard() |> Enum.sort()
 
     entries =
-      for path <- paths do
-        {attrs, body} = parse_contents!(path, File.read!(path), parser_module)
-
-        body =
-          path
-          |> Path.extname()
-          |> String.downcase()
-          |> convert_body(body, opts)
-
-        builder.build(path, attrs, body)
-      end
+      paths
+      |> Enum.map(fn path ->
+        parsed_contents = parse_contents!(path, File.read!(path), parser_module)
+        build_entry(builder, path, parsed_contents, opts)
+      end)
+      |> List.flatten()
```

Comment:
I think we can do something like this:

```suggestion
      Enum.flat_map(paths, fn path ->
        parsed_contents = parse_contents!(path, File.read!(path), parser_module)
        build_entry(builder, path, parsed_contents, opts)
      end)
```

And then you invert on build_entry. Make the tuple return call the list version.

Reply:
Thank you for your guidance. I have submitted the modifications. 

(I am sorry for the delayed response due to the time zone difference.)